### PR TITLE
sdk-zephyr up-merge: 171120 - targeting the dev tag

### PR DESCRIPTION
**Pull request by ioannisg**
Upmerge PR  [**_in-progress_**]

Zephyr Upstream commit: 35264cc214fd37e6da8cc682b4f47e86342f517a

Status:

1. Reverted whatever west ncs-compare indicates (Verified)
2. Reverted more "noup" commits to achieve a conflict-free merge
3. Committed the merge
4. Re-applied reverted patches (not the ones already merged to upstream master)

5. Applied feedback (for more commits to be reverted) by @rlubos and @mbolivar-nordic 

SDK-NRF PR: https://github.com/nrfconnect/sdk-nrf/pull/3357

### CODE REVIEW

**Comment by rlubos** [comment]
I suggest squashing 2900b82, 4e6b1a2 and ce16c8c (my bad I merged this w/o sauce!) into 0d1a6a8. They're all related and it would make future upmerges easier (less reverting).

764308f doesn't seem to be needed - it actually puts `hal_nordic` behind, shouldn't be cherry-picked after revert.

**Comment by rlubos** [comment]
These could also be reverted I believe:
c20fb78d721f0f71815bbbcb9d9610316871d458 was merged in https://github.com/zephyrproject-rtos/zephyr/pull/29409
29881722da70d9659b8f457639b79a28b80a16fa was merged in https://github.com/zephyrproject-rtos/zephyr/pull/29406

Otherwise, it seems we're good (unless I missed something). There are a few `[nrf toup]` that don't seem to be present in upstream yet, two `[nrf temphack]`, and one [nrf fromlist] remaining, that still has PR upstream (https://github.com/zephyrproject-rtos/zephyr/pull/29123, this should also get rid of the 802.15.4 temphack).

**Comment by de-nordic** [comment]
The 06aedc43235ca096a9ce3f8b3f32e0bc0ba47e12 updates ELM FAT  do you have upmerge for that?
There are also several updates of mcumgr revision.

**Comment by ioannisg** [comment]
> I suggest squashing [2900b82](https://github.com/nrfconnect/sdk-zephyr/commit/2900b82769e9f42f1a9ee24f52018730ea37d1c7), [4e6b1a2](https://github.com/nrfconnect/sdk-zephyr/commit/4e6b1a2258a5b61d169b6912aef86ece8f3fe16a) and [ce16c8c](https://github.com/nrfconnect/sdk-zephyr/commit/ce16c8c8dce65c95b0b00874f01540712b075b82) (my bad I merged this w/o sauce!) into [0d1a6a8](https://github.com/nrfconnect/sdk-zephyr/commit/0d1a6a81c03b6e8248558152f8f52eb677e41cb4). They're all related and it would make future upmerges easier (less reverting).
> 
> [764308f](https://github.com/nrfconnect/sdk-zephyr/commit/764308f8ebcffe161a63f5316e74ee99de6a7d76) doesn't seem to be needed - it actually puts `hal_nordic` behind, shouldn't be cherry-picked after revert.

@rlubos Agreed about squashing: I was planning to do it but I wanted you to have an exact view of what was re-applied.

Right about hal-nordic SHA. Makes no sense to use an outdated hal-nordic pointer.

@mbolivar-nordic this is perhaps a feedback for you - we do not need to revert and/or re-apply manifest updates if zephyr itself brings in a newer pointer for hal-nordic or any other module. For future :)

**Comment by anangl** [comment]
@ioannisg I think you should also revert these two commits pointed out by @rlubos above:

> These could also be reverted I believe:
> [c20fb78](https://github.com/nrfconnect/sdk-zephyr/commit/c20fb78d721f0f71815bbbcb9d9610316871d458) was merged in [zephyrproject-rtos/zephyr#29409](https://github.com/zephyrproject-rtos/zephyr/pull/29409)
> [2988172](https://github.com/nrfconnect/sdk-zephyr/commit/29881722da70d9659b8f457639b79a28b80a16fa) was merged in [zephyrproject-rtos/zephyr#29406](https://github.com/zephyrproject-rtos/zephyr/pull/29406)

**Comment by ioannisg** [comment]
> @ioannisg I think you should also revert these two commits pointed out by @rlubos above:
> 
> > These could also be reverted I believe:
> > [c20fb78](https://github.com/nrfconnect/sdk-zephyr/commit/c20fb78d721f0f71815bbbcb9d9610316871d458) was merged in [zephyrproject-rtos/zephyr#29409](https://github.com/zephyrproject-rtos/zephyr/pull/29409)
> > [2988172](https://github.com/nrfconnect/sdk-zephyr/commit/29881722da70d9659b8f457639b79a28b80a16fa) was merged in [zephyrproject-rtos/zephyr#29406](https://github.com/zephyrproject-rtos/zephyr/pull/29406)

Yes @anangl that's the next thing to do, hang on.

**Comment by ioannisg** [comment]
> > @ioannisg I think you should also revert these two commits pointed out by @rlubos above:
> > > These could also be reverted I believe:
> > > [c20fb78](https://github.com/nrfconnect/sdk-zephyr/commit/c20fb78d721f0f71815bbbcb9d9610316871d458) was merged in [zephyrproject-rtos/zephyr#29409](https://github.com/zephyrproject-rtos/zephyr/pull/29409)
> > > [2988172](https://github.com/nrfconnect/sdk-zephyr/commit/29881722da70d9659b8f457639b79a28b80a16fa) was merged in [zephyrproject-rtos/zephyr#29406](https://github.com/zephyrproject-rtos/zephyr/pull/29406)
> 
> Yes @anangl that's the next thing to do, hang on.

@anangl @rlubos the commits are now reverted (before the merge commit)

**Comment by ioannisg** [comment]
@mbolivar-nordic I re-applied one of the reverted commits, as per your suggestion. See 
300dba8

**Review by mbolivar-nordic — APPROVED** [review]
Looks good to me as of 

west ncs-loot says:

```
NCS commit (manifest-rev): e2211715846787786009027375202b85d5619797, upstream commit: 35264cc214fd37e6da8cc682b4f47e86342f517a
OOT patches: 20 total
- 39f7253a986e3b8cb9ef0c8935a2b32c59d8b5f4 [nrf noup] bluetooth: controller: default to SPLIT architecture
- 19c66e05d970234e84f8d8571e0e33a8a5ba7fe4 [nrf noup] net: lib: mqtt: Provide option to enable TLS ession caching
- 45958693f28fb7cdf0b5414e10b9ce7e2a81ce59 [nrf noup] dts: add compatibles for CC310 NS access
- 9c42b9b13d974b7a56642004544b2fc588099693 [nrf noup] dts: prefer CC310 for zephyr,entropy
- 324b6fbb139a69a366ae0f884bda589c9e8e4ffa [nrf noup] codeowner: Add entry for .github and Jenkinsfile
- 23df4b66d44ecc20974598c41bb095995e12def0 [nrf noup] ci: Add Jenkins-based CI for zephyr
- 1a64079c8fe3c0b42f5888abe0d774ec2cb74fba [nrf noup] ci: move jenkinsfile to test-ci-nrfconnect-cfg
- 373acf80e0a3a743e9bfc9ab5d0e8a4e6a56c6b1 [nrf noup] net: l2: Kconfig: Add option for Zigbee L2 layer
- d875660a86418497a90e12185914eae902105e4c [nrf toup] drivers: clock_control: nrf: Change CLOCK_CONTROL_NRF_FORCE_A
- 7c4e6bb7958a62e4666fde529c6f7e77d6a44d04 [nrf toup] drivers: clock_control: nrf: Add assert on unexpected event
- a32dc3c16743280c5b4042a5b3b41034631b78e4 [nrf toup] drivers: clock_control: nrf: Add CLOCK_CONTROL_NRF_FORCE_ALT
- 3ac213503417bd1f26b3fda4e443d1a91877a8b9 [nrf toup] bintools: append ; to > to ensure a space
- b38a27ce157502c67d365e7e7caa43cdfe47e578 [nrf fromlist] drivers: ieee802154: nrf5: Add support for tx in the future
- 0f3124de75a0159c5011ef520cc3fe7d47ed37f6 [nrf toup] drivers: Adding cryptocell as chosen entropy for nRF5340
- d46a1ebff09d4d64bb7ec21423a61d6d91813e97 [nrf temphack] drivers: ieee802154: nrf5: Remove tx_started callback
- 06f3f56d5dc48e26d8b5937ac9167fd3f536f569 [nrf temphack] soc: use PM when configuring MPU regions
- 03934abe053a8abf1f4b8fadc00e2de9e8aec88a [nrf noup] include: net: add NCS extensions
- df7f27dbd88a1b48a65782c62ef06cb1375d02c4 [nrf noup] tree-wide: support NCS Partition Manager (PM) definitions
- a45aa8b78b7e23cfcbcd01ad5212c9f498f6a0dc [nrf noup] linker: use partition manager to set RAM
- e2211715846787786009027375202b85d5619797 [nrf noup] doc: Share Kconfig documentation between repos
```

I can cherry-pick those clean on top of the latest upstream merge-base (35264cc214) and get a zero-diff linear history, which I've pushed here (this is just for reference, don't take this tree into this PR):

https://github.com/mbolivar-nordic/fw-nrfconnect-zephyr/tree/upmerge-389-rebase

**Review by anangl — APPROVED** [review]
LGTM

**Review by rlubos — APPROVED** [review]

**Comment by thst-nordic** [comment]
Give me a bit to figure out the CI changes.

**Review by tejlmand — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Comment by ioannisg** [comment]
Patch by @hakonfam is in accordance with an upstream PR for clearing the SPLIM regs early enough.

**Review by de-nordic — APPROVED** [review]
Did some testing on subsys/fs and mcumgr, seems OK.

**Comment by ioannisg** [comment]
Scan code failure seems relevant to upstream patches (xtensa), so we ignore?
How about the Device tree failure?

**Comment by thst-nordic** [comment]
I disabled ScanCode. 
'Device tree' is deprecated and will disappear on next commit.

**Comment by ioannisg** [comment]
> Patch by @hakonfam is in accordance with an upstream PR for clearing the SPLIM regs early enough. It will be merged independently.

No, my bad, I forgot this was actually pushed on top, here. Ignore.

**Review by joerchan — APPROVED** [review]

File: subsys/dfu/boot/mcuboot.c [hunk 1]
```diff
@@ -77,7 +77,9 @@ struct mcuboot_v1_raw_header {
 #define FLASH_AREA_IMAGE_SECONDARY FLASH_AREA_ID(image_1_nonsecure)
 #else
 #define FLASH_AREA_IMAGE_PRIMARY FLASH_AREA_ID(image_0)
+#if FLASH_AREA_LABEL_EXISTS(image_1)
```
Comment (de-nordic) [inline subsys/dfu/boot/mcuboot.c:80]: This probably causes problem with `smp_svr` overwriting itself. The compiler does not go inside the `#if`, probably due to the PM taking over the role of flash manager.

Comment (hakonfam) [reply subsys/dfu/boot/mcuboot.c:80]: I will look into it

Comment (hakonfam) [reply subsys/dfu/boot/mcuboot.c:80]: https://github.com/ioannisg/sdk-nrf/pull/2

Can you try this?

Comment (ioannisg) [reply subsys/dfu/boot/mcuboot.c:80]: @de-nordic can you confirm?

**Review by hakonfam — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
Traceback (most recent call last):
      File "/home/work/modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot/scripts/wrapper/wrapper.py", line 20, in <module>
        import imgtool
    ModuleNotFoundError: No module named 'imgtool'

https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-zephyr/job/PR-389/lastCompletedBuild/testReport/nrf9160dk_nrf9160ns_samples_tfm_integration_psa_level_1_sample/psa_level_1/Execution___SanityCheckNRF_gnuarmemb_4_4___Sanity_Check___samples_tfm_integration_psa_level_1_sample_psa_level_1/

**Comment by ioannisg** [comment]
> ```
> Traceback (most recent call last):
>   File "/home/work/modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot/scripts/wrapper/wrapper.py", line 20, in <module>
>     import imgtool
> ModuleNotFoundError: No module named 'imgtool'
> ```
> 
> https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-zephyr/job/PR-389/lastCompletedBuild/testReport/nrf9160dk_nrf9160ns_samples_tfm_integration_psa_level_1_sample/psa_level_1/Execution___SanityCheckNRF_gnuarmemb_4_4___Sanity_Check___samples_tfm_integration_psa_level_1_sample_psa_level_1/

@thst-nordic FYI 
Do you think imgtool is missing?

**Comment by thst-nordic** [comment]
> ```
> Traceback (most recent call last):
>   File "/home/work/modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot/scripts/wrapper/wrapper.py", line 20, in <module>
>     import imgtool
> ModuleNotFoundError: No module named 'imgtool'
> ```
> 
> https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-zephyr/job/PR-389/lastCompletedBuild/testReport/nrf9160dk_nrf9160ns_samples_tfm_integration_psa_level_1_sample/psa_level_1/Execution___SanityCheckNRF_gnuarmemb_4_4___Sanity_Check___samples_tfm_integration_psa_level_1_sample_psa_level_1/

I have installed imgtool,  it will be installed on next run of CI.   
but the target is still failing with 2 missing files: 

/work/zephyr/samples/tfm_integration/psa_level_1/src/main.c:11:10: fatal error: tfm_ns_interface.h: No such file or directory
   11 | #include "tfm_ns_interface.h"
      |          ^~~~~~~~~~~~~~~~~~~~

/work/zephyr/samples/tfm_integration/psa_level_1/src/psa_attestation.c:11:10: fatal error: psa/initial_attestation.h: No such file or directory
   11 | #include "psa/initial_attestation.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~

**Comment by ioannisg** [comment]
> > ```
> > Traceback (most recent call last):
> >   File "/home/work/modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot/scripts/wrapper/wrapper.py", line 20, in <module>
> >     import imgtool
> > ModuleNotFoundError: No module named 'imgtool'
> > ```
> > 
> > 
> > https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-zephyr/job/PR-389/lastCompletedBuild/testReport/nrf9160dk_nrf9160ns_samples_tfm_integration_psa_level_1_sample/psa_level_1/Execution___SanityCheckNRF_gnuarmemb_4_4___Sanity_Check___samples_tfm_integration_psa_level_1_sample_psa_level_1/
> 
> I have installed imgtool, it will be installed on next run of CI.
> but the target is still failing with 2 missing files:
> 
> /work/zephyr/samples/tfm_integration/psa_level_1/src/main.c:11:10: fatal error: tfm_ns_interface.h: No such file or directory
> 11 | #include "tfm_ns_interface.h"
> | ^~~~~~~~~~~~~~~~~~~~
> 
> /work/zephyr/samples/tfm_integration/psa_level_1/src/psa_attestation.c:11:10: fatal error: psa/initial_attestation.h: No such file or directory
> 11 | #include "psa/initial_attestation.h"
> | ^~~~~~~~~~~~~~~~~~~~~~~~~~~

@oyvindronningstad ^^

**Comment by oyvindronningstad** [comment]
I can't find the logs for the error, where are they? Did you rerun CI after the imgtool fix?

Anyways, the Zephyr TFM samples are broken in any case, until https://github.com/nrfconnect/sdk-nrf/pull/3306 is in, since SPM is added for all ns boards, conflicting with TFM.

**Comment by ioannisg** [comment]
> I can't find the logs for the error, where are they? Did you rerun CI after the imgtool fix?
> 
> Anyways, the Zephyr TFM samples are broken in any case, until [nrfconnect/sdk-nrf#3306](https://github.com/nrfconnect/sdk-nrf/pull/3306) is in, since SPM is added for all ns boards, conflicting with TFM.

This is a problem. This means we need to have an exception, or merge the two PRs together...

**Review by trond-snekvik — APPROVED** [review]

**Comment by oyvindronningstad** [comment]
> 
> 
> > I can't find the logs for the error, where are they? Did you rerun CI after the imgtool fix?
> > Anyways, the Zephyr TFM samples are broken in any case, until [nrfconnect/sdk-nrf#3306](https://github.com/nrfconnect/sdk-nrf/pull/3306) is in, since SPM is added for all ns boards, conflicting with TFM.
> 
> This is a problem. This means we need to have an exception, or merge the two PRs together...

It seems if we merge https://github.com/nrfconnect/sdk-nrf/pull/3268 and cherry-pick https://github.com/nrfconnect/sdk-nrf/pull/3306/commits/dc0bb441fba74ed6283a3bcbbda902f41389879b, the Zephyr tfm samples build and run fine (when flashed the Zephyr way).

**Comment by ioannisg** [comment]
Build is green now.

**Pull request merged**